app/jobs/imo/invite_patient.rb

```diff
@@ -7,6 +7,12 @@ def perform(patient_id)
     return unless Flipper.enabled?(:imo_messaging)
 
     patient = Patient.find(patient_id)
-    ImoApiService.new.send_invitation(patient)
+    status = ImoApiService.new.send_invitation(patient)
+
+    if patient.imo_authorization
+      patient.imo_authorization.update!(status: status, last_invited_at: Time.current)
+    else
+      ImoAuthorization.create!(patient: patient, status: status, last_invited_at: Time.current)
```

Comment:
I originally had the imo auth happen here but moved it out for reasons I don't recall. That was a mistake that's bothered me ever since, so I'm moving it back to where it belongs and we have better separation of concerns.